src/nvim/getchar.c

```diff
@@ -4261,3 +4261,21 @@ static bool typebuf_match_len(const uint8_t *str, int *mlen)
   *mlen = i;
   return str[i] == NUL;  // matched the whole string
 }
+
+mapblock_T * get_maphash(int index, char *scope)
+{
+  if (index > MAX_MAPHASH) {
+    return NULL;
+  }
+
+  bool buffer_maps = (STRNCMP(scope, "buffer", 6) == 0);
+  bool global_maps = (STRNCMP(scope, "global", 6) == 0);
+
+  if (buffer_maps) {
```

Comment:
Though why `STRNCMP`? This makes arguments like `globalXXX` valid. Should be `strcmp`: no macros (macros was created by Bram only because he uses `char_u`, this function uses `char`), no `n`.